doc/Developing.md

```diff
@@ -0,0 +1,124 @@
+  * WDL test files belong in the top directory `test`
+* Submit a pull request when you are ready to have your code reviewed.
+
+### Style guidelines
+
+* We use [scalafmt style](https://scalameta.org/scalafmt/) with a few modifications. You don't need to worry so much about code style since you will use the automatic formatter on your code before it is committed.
+* Readability is more important than efficiency or concision - write more/slower code if it makes the code more readable.
+* Avoid using more complex features, e.g. reflection.
+
+## Using sbt
+
+sbt is the build system for Scala (similar to a Makefile). The following are the main commands you will use while developing.
+
+### Compiling the code
+
+Scala (like Java) is a compiled language. To compile, run:
+
+```
+$ sbt compile
+```
+
+If there are errors in your code, the compiler will fail with (hopefully useful) error messages.
+
+### Running unit tests
+
+You should always run the unit tests after every successful compile. Generally, you want to run `sbt testQuick`, which only runs the tests that failed previously, as well as the tests for any code you've modified since the last time you ran the tests. However, the first time you checkout the code (to make sure your development environment is set up correctly) and then right before you push any changes to the repository, you should run the full test suite using `sbt test`.
+
+### Running the integration tests
+
+Integration tests actually build and run apps/workflows on DNAnexus. These tests take much longer to run than the unit tests, and so typically you only run them before submitting a pull request.
+
+First, you need to an account on the DNAnexus staging environment, and you need to be added to the projects that have been setup for testing; @orodeh can set this up for you. Next, log into the DNAnexus staging environment using dx-toolkit: `dx login --staging`. Note that very often your login will timeout while the integration tests are running unless you are actively using the platform in another session, and this will cause the tests to fail. To avoid, this, generate a token via the web UI and use that token to log in on the command line: `dx login --staging --token <token>`.
+
+Next, delete any existing build artifacts using the following commands:
+
+```
+$ sbt clean
+$ sbt cleanFiles
+$ find . -name target | xargs rm -rf
+```
+
+### Other sbt tips
+
+#### Cache
+
+sbt keeps the cache of downloaded jar files in `${HOME}/.ivy2/cache`. For example, the WDL jar files are under `${HOME}/.ivy2/cache/org.broadinstitute`. In case of problems with cached jars, you can remove this directory recursively. This will make WDL download all dependencies (again).
+
+## Releasing a new version
+
+### Release check list
+
+- Make sure regression tests pass
+- Update release notes and README.md
+- Make sure the version number in `src/main/resources/application.conf` is correct. It is used
```

Comment:
We don't currently have such a file, but you can add it. It would be a good idea.